### DISCUSSION TITLE
Blazor and CriteriaBase

### DISCUSSION THREAD

**Discussion by mikeclimbs23** [MarimerLLC/csla#2236] [Questions]
I was reading through the discussion board and noticed a mention that CriteriaBase has been depreciated for use with CommandBase. Is that also the case for ReadOnlyListBase?

We're using ReadOnlyBase to execute stored procedures that could have as many as 10 optional parameters. I think it would be easier to read the code using a criteria base object rather than a procedure call with 10 parameters.

Thanks,

Mike

**Comment by rockfordlhotka** [comment]
CriteriaBase is deprecated in general, because the data portal now allows passing arbitrary (serializable) parameters for create and fetch operations.

CommandBase can have properties of any serializable type. The idea with CommandBase is to create an instance of your command object, set its properties, and execute the command. You may have used property of a type derived from CriteriaBase in the past, but that is really no different than ReadOnlyBase, and the readonly stereotypes remain very important.
